osu.Game/Overlays/BeatmapSet/LeaderboardModSelector.cs

```diff
@@ -0,0 +1,144 @@
+    {
+        public Bindable<IEnumerable<Mod>> SelectedMods = new Bindable<IEnumerable<Mod>>();
+
+        private RulesetInfo ruleset = new RulesetInfo();
+        private readonly FillFlowContainer<SelectableModIcon> modsContainer;
+
+        public LeaderboardModSelector()
+        {
+            AutoSizeAxes = Axes.Y;
+            RelativeSizeAxes = Axes.X;
+            Child = modsContainer = new FillFlowContainer<SelectableModIcon>
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                AutoSizeAxes = Axes.Both,
+                Direction = FillDirection.Full,
+                Spacing = new Vector2(4),
+            };
+        }
+
+        public void ResetRuleset(RulesetInfo ruleset)
+        {
+            SelectedMods.Value = new List<Mod>();
+
+            if (ruleset == null)
+            {
+                this.ruleset = ruleset;
+                modsContainer.Clear();
+                return;
+            }
+
+            if (this.ruleset?.Equals(ruleset) ?? false)
+            {
+                deselectAll();
+                return;
+            }
+
+            this.ruleset = ruleset;
+
+            modsContainer.Clear();
+
+            foreach (var mod in ruleset.CreateInstance().GetAllMods())
+            {
+                if (mod.Ranked)
+                    modsContainer.Add(new SelectableModIcon(mod));
+            }
+
+            foreach (var mod in modsContainer)
+                mod.OnSelectionChanged += selectionChanged;
+        }
+
+        private void selectionChanged(Mod mod, bool selected)
+        {
+            var mods = SelectedMods.Value?.ToList() ?? new List<Mod>();
+
+            if (selected)
+                mods.Add(mod);
+            else
+                mods.Remove(mod);
+
+            SelectedMods.Value = mods;
+        }
+
+        private void deselectAll()
+        {
+            foreach (var mod in modsContainer)
+                mod.Selected.Value = false;
+        }
+
+        private class SelectableModIcon : Container
+        {
+            private const float mod_scale = 0.4f;
+            private const int duration = 200;
+
+            public readonly BindableBool Selected = new BindableBool(false);
+            public Action<Mod, bool> OnSelectionChanged;
+
+            private readonly ModIcon modIcon;
+            private readonly Mod mod;
+
+            public SelectableModIcon(Mod mod)
+            {
+                this.mod = mod;
+
+                AutoSizeAxes = Axes.Both;
+                Children = new Drawable[]
+                {
+                    modIcon = new ModIcon(mod)
+                    {
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Scale = new Vector2(mod_scale),
+                    },
+                    new HoverClickSounds()
+                };
+
+                Selected.BindValueChanged(_ => updateState());
```

Comment:
```suggestion
                Selected.BindValueChanged(_ => updateState(), true);
```
runs once immediately without needing to do `TriggerChange`